src/charm.py

```diff
@@ -142,6 +169,53 @@ def _run_sql_migration(self, set_timeout: bool) -> None:
         stdout, _ = process.wait_output()
         logger.info(f"Executing automigration: {stdout}")
 
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Hydra container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Hydra container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring resources")
+
+        if not self._hydra_service_is_running:
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for Hydra service")
+            logger.info("Hydra service is absent. Deferring the event.")
+            return
+
+        if not self.model.relations[self._db_relation_name]:
+            self.unit.status = BlockedStatus("Missing required relation with postgresql")
+            return
+
+        if not self.database.is_resource_created():
+            self.unit.status = WaitingStatus("Waiting for database creation")
+            event.defer()
+            return
+
+        self._container.push(self._hydra_config_path, self._render_conf_file(), make_dirs=True)
+        self._container.start(self._container_name)
+        self.unit.status = ActiveStatus()
+
+    def _update_hydra_endpoints_relation_data(self, event) -> None:
+        admin_endpoint = (
+            self.admin_ingress.url
+            if self.model.relations["admin-ingress"]
+            else f"{self.app.name}.{self.model.name}.svc.cluster.local:{HYDRA_ADMIN_PORT}",
+        )
+        public_endpoint = (
+            self.public_ingress.url
+            if self.model.relations["public-ingress"]
+            else f"{self.app.name}.{self.model.name}.svc.cluster.local:{HYDRA_PUBLIC_PORT}",
+        )
+        self.endpoints_provider.send_endpoint_relation_data(
+            self.app, admin_endpoint[0], public_endpoint[0]
+        )
+
+        logger.info(
+            f"Sent endpoints info: public - {public_endpoint[0]} admin - {admin_endpoint[0]}"
+        )
+
     def _on_hydra_pebble_ready(self, event: WorkloadEvent) -> None:
```

Comment:
Is there a reason why pebble ready calls this same event handler `_handle_status_update_config` it seems to be doing the exact same thing right?

Reply:
You could simply add the pebble layer part to `_handle_status_update_config` ex.
```
if the existing pebble layer is not the same as `pebble_layer`:
    add pebble layer
restart service
```